### ISSUE TITLE
make test fails in osx due to mktemp

### ISSUE THREAD

**Issue by ashee** [docker/hyperkit#52]
```
amitava:hyperkit amitava$ make
gen src/dtrace.d
cc src/vmm/x86.c
cc src/vmm/vmm.c
cc src/vmm/vmm_host.c
cc src/vmm/vmm_mem.c
cc src/vmm/vmm_lapic.c
cc src/vmm/vmm_instruction_emul.c
cc src/vmm/vmm_ioport.c
cc src/vmm/vmm_callout.c
cc src/vmm/vmm_stat.c
cc src/vmm/vmm_util.c
cc src/vmm/vmm_api.c
cc src/vmm/intel/vmx.c
cc src/vmm/intel/vmx_msr.c
cc src/vmm/intel/vmcs.c
cc src/vmm/io/vatpic.c
cc src/vmm/io/vatpit.c
cc src/vmm/io/vhpet.c
cc src/vmm/io/vioapic.c
cc src/vmm/io/vlapic.c
cc src/vmm/io/vpmtmr.c
cc src/vmm/io/vrtc.c
cc src/acpitbl.c
cc src/atkbdc.c
cc src/block_if.c
cc src/consport.c
cc src/dbgport.c
cc src/inout.c
cc src/ioapic.c
cc src/md5c.c
cc src/mem.c
cc src/mevent.c
cc src/mptbl.c
cc src/pci_ahci.c
cc src/pci_emul.c
cc src/pci_hostbridge.c
cc src/pci_irq.c
cc src/pci_lpc.c
cc src/pci_uart.c
cc src/pci_virtio_9p.c
cc src/pci_virtio_block.c
cc src/pci_virtio_net_tap.c
cc src/pci_virtio_net_vmnet.c
cc src/pci_virtio_net_vpnkit.c
cc src/pci_virtio_rnd.c
cc src/pm.c
cc src/post.c
cc src/rtc.c
cc src/smbiostbl.c
cc src/task_switch.c
cc src/uart_emul.c
cc src/xhyve.c
cc src/virtio.c
cc src/xmsr.c
cc src/firmware/bootrom.c
cc src/firmware/kexec.c
cc src/firmware/fbsd.c
ld com.docker.hyperkit.sym
dsym com.docker.hyperkit.dSYM
strip com.docker.hyperkit
```
Run test - fails
```
amitava:hyperkit amitava$ make test
usage: mktemp [-d] [-q] [-t prefix] [-u] template ...
       mktemp [-d] [-q] [-u] -t prefix
make: *** [test/vmlinuz] Error 1
```
Generalize mktemp call - see http://unix.stackexchange.com/questions/30091/fix-or-alternative-for-mktemp-in-os-x
```diff
amitava:hyperkit amitava$ git diff HEAD^ HEAD
diff --git a/test/tinycore.sh b/test/tinycore.sh
index c3eff43..67f3de1 100755
--- a/test/tinycore.sh
+++ b/test/tinycore.sh
@@ -9,7 +9,7 @@ set -e

 BASE_URL="http://distro.ibiblio.org/tinycorelinux/"

-TMP_DIR=$(mktemp -d)
+TMP_DIR=$(mktemp -q -d -t "$(basename "$0").XXXXXX")=
 INITRD_DIR="${TMP_DIR}"/initrd
```
Now getting Error 23
```
amitava:hyperkit amitava$ make test
Downloading tinycore linux
make: *** [test/initrd.gz] Error 23
```

**Comment by ijc25** [comment]
The current `make test` works in CI and on my local system where I am using the system `/usr/bin/mktemp`. This version of `mktemp` seems completely happy with not being given a template.

Please can you provide more information about your system and in particular where/why you `mktemp` might differ from the normal one. Perhaps you running a different (older?) version of OSX? (I'm running 10.11.6, I'm not sure what CI is running).

Lets take things one step at a time ignore the follow on `Error 23` which you got after modifying the code for now.

**Comment by ashee** [comment]
OSX Yosemite 
Version 10.10.5 (14F1909)
```
amitava:~ amitava$ uname -a
Darwin amitava.local 14.5.0 Darwin Kernel Version 14.5.0: Thu Jun 16 19:58:21 PDT 2016; root:xnu-2782.50.4~1/RELEASE_X86_64 x86_64
```
```
amitava:~ amitava$ echo $SHELL
/bin/bash
amitava:~ amitava$ echo $BASH_VERSION
3.2.57(1)-release
```

**Comment by ijc25** [comment]
Please can you post the output of all three of the following on your system:

* `mktemp -d`
* `mktemp -d tmp.XXXXXX`
* `mktemp -t tmp`

**Comment by ashee** [comment]
```
amitava:~ amitava$ mktemp -d
usage: mktemp [-d] [-q] [-t prefix] [-u] template ...
       mktemp [-d] [-q] [-u] -t prefix
amitava:~ amitava$ mktemp -d tmp.XXXXXX
tmp.O54KP1
amitava:~ amitava$ mktemp -t tmp
/var/folders/fn/2f_klz19237bvm89n2sns3d80000gp/T/tmp.tvF7iWiu
```

**Comment by ijc25** [comment]
Thanks. Looks like the correct construct to use then is the last one. I'd suggest:
```
-TMP_DIR=$(mktemp -d)
+TMP_DIR=$(mktemp -d -t hyperkit)
```
Care to make a PR?

**Comment by ijc25** [comment]
Closed via #53, thanks
